projects/onto-search/src/lib/onto-search-facet/range-facet/range-histogram/range-histogram.component.ts

```diff
@@ -0,0 +1,129 @@
+import {AfterViewInit, Component, ElementRef, Input, ViewChild} from '@angular/core';
+import {SearchFacetModel} from '../../models/search-facet-model';
+import {HistogramModel} from '../models/histogram-model';
+import {SearchRangeFacetGroupModel} from '../models/search-range-facet-group-model';
+
+@Component({
+  selector: 'onto-range-histogram',
+  templateUrl: './range-histogram.component.html',
+  styleUrls: ['./range-histogram.component.scss']
+})
+export class RangeHistogramComponent implements AfterViewInit {
+  /**
+   * Facet data holder.
+   */
+  @Input()
+  public data: SearchRangeFacetGroupModel;
+
+  @ViewChild('rangeHistogramCanvas')
+  private canvas: ElementRef;
+
+  private ctx;
+  private histogramModel: HistogramModel;
+
+  public ngAfterViewInit(): void {
+    this.histogramModel = this.data.histogramConfiguration;
+    this.init();
+
+    // An observer that monitors the resizing of the canvas element, causing it to redraw on changes.
+    // @ts-ignore
+    const observer = new ResizeObserver(() => {
+      this.init();
+    });
+
+    observer.observe(this.canvas.nativeElement);
+  }
+
+  init(): void {
+    this.fitToContainer(this.canvas.nativeElement);
+    this.ctx = this.canvas.nativeElement.getContext('2d');
+
+    const data: SearchFacetModel[] = [...this.data.facetGroupData];
+
+    data.sort((a, b) => {
+      try {
```

Comment:
This try-catch is useless. You will never have an error reaching here given you handle the parsing errors in the parseInt method below.